### PR TITLE
Password hashing and checking functionality with associated unit tests

### PR DESCRIPTION
**Pull request by la289**
Small first commit just to iron out kinks. 

This branch includes the password hashing utilities to both set a users password and check it after the fact.

### CODE REVIEW

File: utils/passhash.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	"golang.org/x/crypto/bcrypt"
+)
+
+func GeneratePassHash(password string) ([]byte, error) {
+	// could vaildate password requirements, but since we are not
+	// implementing a create account, leaving it out for now
+	return bcrypt.GenerateFromPassword([]byte(password), 10)
+}
+
+func CheckPassword(hashedpass []byte, password string) error {
```
Comment (awly) [inline utils/passhash.go:13]: nit: why use different argument types?

File: utils/utils_test.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package utils
+
+import (
+	"testing"
+)
+
+func TestGeneratePassHash(t *testing.T) {
```
Comment (awly) [inline utils/utils_test.go:7]: this test is essentially testing the `bcrypt` package.
leave it to that package's authors to test their implementation and remove this test

File: utils/utils_test.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	}
+
+	for _, c := range cases {
+		hp, err := GeneratePassHash(c.pass)
+		if err != nil {
+			t.Errorf("Generating password hash failed.")
+		}
+		err = CheckPassword(hp, c.pass)
+		if err != nil {
+			t.Errorf("Password Comparison failed.")
```
Comment (awly) [inline utils/utils_test.go:42]:
```suggestion
			t.Errorf("Generating password hash failed: %v", err)
		}
		err = CheckPassword(hp, c.pass)
		if err != nil {
			t.Errorf("Password Comparison failed: %v", err)
```

File: utils/utils_test.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+		}
+	}
+}
+
+func TestCheckPasswordFail(t *testing.T) {
```
Comment (awly) [inline utils/utils_test.go:47]: same as above, this is basically testing the `bcrypt` package

**Review by awly — APPROVED** [review]

File: utils/passhash.go [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package utils
+
+import(
+	"golang.org/x/crypto/bcrypt"
+)
+
+func GeneratePassHash(password string) ([]byte, error) {
+	// could vaildate password requirements, but since we are not
```
Comment (r0mant) [inline utils/passhash.go:8]: validate

File: utils/passhash.go [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+	// implementing a create account, leaving it out for now
+	return bcrypt.GenerateFromPassword([]byte(password), 10)
+}
+
+func CheckPassword(hashedpass []byte, password string) error {
+	return bcrypt.CompareHashAndPassword(hashedpass, []byte(password))
```
Comment (r0mant) [inline utils/passhash.go:14]: Is there really any need to have these two functions if both are just 1-line wrappers over respective bcrypt methods? Just use them directly in the code.

**Review by r0mant — COMMENTED** [review]
Mostly mirroring Andrew's feedback, I'm not sure why these two functions are needed and the tests seem to be basically testing bcrypt.

**Pull request closed without merging**